src/Http/Headers/test/MediaTypeHeaderValueTest.cs

```diff
@@ -683,6 +683,83 @@ public void TryParseStrictList_WithSomeInvalidValues_ReturnsFalse()
             Assert.False(MediaTypeHeaderValue.TryParseStrictList(inputs, out var results));
         }
 
+        [Theory]
+        [InlineData("*/*;", "*/*")]
+        [InlineData("text/*", "text/*")]
+        [InlineData("*/*;", "text/plain")]
+        [InlineData("text/plain", "text/plain")]
+        [InlineData("text/plain;", "text/plain")]
+        [InlineData("text/plain;", "TEXT/PLAIN")]
+        [InlineData("application/json;v=2", "application/json;param=4")]
+        public void MatchesMediaType_PositiveCases(string mediaType1, string mediaType2)
+        {
+            // Arrange
+            var parsedMediaType1 = MediaTypeHeaderValue.Parse(mediaType1);
+
+            // Act
+            var matches = parsedMediaType1.MatchesMediaType(mediaType2);
+
+            // Assert
+            Assert.True(matches);
+        }
+
+        [Theory]
+        [InlineData("application/html", "text/*")]
+        [InlineData("application/json", "application/html")]
+        [InlineData("text/plain;", "*/*")]
+        public void MatchesMediaType_NegativeCases(string mediaType1, string mediaType2)
+        {
+            // Arrange
+            var parsedMediaType1 = MediaTypeHeaderValue.Parse(mediaType1);
+
+            // Act
+            var matches = parsedMediaType1.MatchesMediaType(mediaType2);
+
+            // Assert
+            Assert.False(matches);
+        }
+
+        [Theory]
+        [InlineData("application/entity+json", "application/entity+json")]
+        [InlineData("application/*+json", "application/entity+json")]
+        [InlineData("application/*+json", "application/*+json")]
+        [InlineData("application/problem+json", "application/json")]
+        [InlineData("application/vnd.restful+json", "application/json")]
+        [InlineData("application/*", "application/*+JSON")]
+        [InlineData("application/*", "application/entity+JSON")]
+        [InlineData("*/*", "application/entity+json")]
+        public void MatchesMediaTypeWithSuffixes_PositiveCases(string mediaType1, string mediaType2)
+        {
+            // Arrange
+            var parsedMediaType1 = MediaTypeHeaderValue.Parse(mediaType1);
+
+            // Act
+            var result = parsedMediaType1.MatchesMediaType(mediaType2);
+
+            // Assert
+            Assert.True(result);
```

Comment:
A secondary assertion to add here would be to create a MediaTypeHeaderValue for mediaType2 and assert that you get the same result for parsedMediaType2.IsSubsetOf(parsedMediaType1).

Reply:
I added IsSubsetOf result comparison to all test methods.